### PR TITLE
MAGN-10516 Lacing properties should publish correctly from Studio to Flood/Reach

### PR DESCRIPTION
**Pull request by saintentropy**
### Purpose

This fix ensures backward compatibly of Dynamo.Storage with Dynamo Core.  The LacingStrategy enum needed to be re-ordered back to the original sequence to maintain binary-level compatibility without requiring recompilation.  This change will also fix errors in client code that had referenced the LacingStrategy enum prior to the addition of auto.
    
### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@mjkkirschner 

### FYIs

@ramramps @ke-yu @QilongTang

### CODE REVIEW

File: src/DynamoCore/Graph/Nodes/NodeModel.cs [hunk 1]
```diff
@@ -2216,11 +2216,11 @@ public enum ElementState
     public enum LacingStrategy
     {
         Disabled,
-        Auto,
```
Comment (saintentropy) [inline src/DynamoCore/Graph/Nodes/NodeModel.cs]: The addition of the auto in this location caused the silent failure in client code consuming the API as the enum values are constants and backed into the the client's assembly IL.

See: http://stackoverflow.com/questions/1456785/a-definitive-guide-to-api-breaking-changes-in-net (Thanks @mjkkirschner )

**Comment by mjkkirschner** [comment]
@saintentropy looks good to me - does @jnealb have a regression for this class of defect?

**Comment by jnealb** [comment]
@mjkkirschner @saintentropy LotsOfStuff is a good start

**Comment by ke-yu** [comment]
LGTM

**Comment by QilongTang** [comment]
Wow. Definately good stuff. Thank you guys for getting the right solution.

**Comment by QilongTang** [comment]
Kicking off the job again succeeded. Merging.

**Pull request merged**